Enhance documentation on layouting GLSP diagrams (#56)
eclipsesource/graphical-lsp-website: content/documentation/clientLayouting/content.md

The children of the container are layouted according to their explicit positions. Based on that, the maximum height and width are computed and are used as bounds for the container.

##### Common Layout Options

- Vertical alignment `GConstants.VAlign`: `TOP` | `CENTER` | `BOTTOM`
- Horizontal alignment `GConstants.HAlign`: `LEFT` | `CENTER` | `RIGHT`
- Preferred height / width `GLayoutOptions.KEY_PREF_WIDTH` | `GLayoutOptions.KEY_PREF_HEIGHT`

For more details, please see [`GLayoutOptions`](https://github.com/eclipse-glsp/glsp-server/blob/master/plugins/org.eclipse.glsp.graph/src/org/eclipse/glsp/graph/builder/impl/GLayoutOptions.java)
and [`GConstants`](https://github.com/eclipse-glsp/glsp-server/blob/master/plugins/org.eclipse.glsp.graph/src/org/eclipse/glsp/graph/util/GConstants.java).

#### Custom Layouter

Additional custom layouters can be contributed by creating a layouter class that extends the `AbstractLayouter` and optionally provide custom options that extend the `AbstractLayoutOptions`.

In the following example we show a simple custom layouter that extends the `VBoxLayouterExt` and provides a custom layout option,

> which increases all children sizes by this value in both width and height, the default value is `50px`.

Reviewer: ```suggestion
which increases the widths and heights of all children by a specified value, the default value is `50` pixel.
```

  enlargeSizeBy: number;
}

@injectable()
export class MyCustomLayouter extends VBoxLayouterExt {
  static override KIND = "myCustomLayout";

  protected override getChildrenSize(
    container: SParentElement & LayoutContainer,
    containerOptions: MyCustomLayoutOptions,
    layouter: StatefulLayouter
  ): Dimension {
    const result = super.getChildrenSize(container, containerOptions, layouter);
    return {
      width: result.width + containerOptions.enlargeSizeBy,
      height: result.height + containerOptions.enlargeSizeBy,
    };
  }

  protected override getDefaultLayoutOptions(): MyCustomLayoutOptions {
    return {
      enlargeSizeBy: 50,
      ...super.getDefaultLayoutOptions(),
    };
  }

  protected override spread(
    a: MyCustomLayoutOptions,
    b: MyCustomLayoutOptions
  ): MyCustomLayoutOptions {
    return { ...a, ...b };
  }
}
```

</details>

Then the new custom layout needs to bound in the diagram module (`di.config.ts`):

<details open><summary>GLSP Client</summary>

```ts
const myDiagramModule = new ContainerModule((bind, unbind, isBound, rebind) => {
   rebind(VBoxLayouter).to(MyCustomLayouter);
}
```

</details>

> For independent implementations that extend the `AbstractLayout`, it will be bound as follows:

Reviewer: ```suggestion
Custom layouter, which don't overwrite existing ones should extend the `AbstractLayout` and can be registered for a certain layouter key, e.g. `MyCustomLayouter.KIND`, that is used in the `layout` property of compartments as follows:
```

<details open><summary>GLSP Client</summary>

```ts
const myDiagramModule = new ContainerModule((bind, unbind, isBound, rebind) => {
   configureLayout({ bind, isBound }, MyCustomLayouter.KIND, MyCustomLayouter);
}
```

</details>

> Then the custom layouter is ready to use:

Reviewer: ```suggestion
With that, the custom layouter is ready to be used by compartments in the graphical model:
```
